diff --git a/emscripten/tools/ports/sdl3.py b/emscripten/tools/ports/sdl3.py
index dce32b5..81391a0 100644
--- a/emscripten/tools/ports/sdl3.py
+++ b/emscripten/tools/ports/sdl3.py
@@ -8,10 +8,10 @@ import shutil
 
 from tools import diagnostics
 
-VERSION = '3.4.2'
+VERSION = '3.4.10'
 TAG = f'release-{VERSION}'
-HASH = 'a17fe538993a3956e0b85fda21e7b431244e803a5facb35bb7a2bfd9ee23f1aac65838ed3225f526b81410cae7c23da7c40693c2e791385281f0764239116bce'
-SUBDIR = f'SDL-{TAG}'
+HASH = 'a66366071aa5dcc5949c3f48b2041dff8850cd05917c1ba5f6ae251e9eef3277d31884fe020251388051887c8c1587cd80457a02bdb96c11eb41065ac9f69958'
+SUBDIR = f'SDL3-{VERSION}'
 
 variants = {'sdl3-mt': {'PTHREADS': 1}}
@@ -29,7 +29,7 @@ def get(ports, settings, shared):
   diagnostics.warning('experimental', 'sdl3 port is still experimental')
 
   # get the port
-  ports.fetch_project('sdl3', f'https://github.com/libsdl-org/SDL/archive/{TAG}.zip', sha512hash=HASH)
+  ports.fetch_project('sdl3', f'https://github.com/libsdl-org/SDL/releases/download/{TAG}/SDL3-{VERSION}.tar.gz', sha512hash=HASH)
 
   def create(final):
     root_dir = ports.get_dir('sdl3', SUBDIR)
